src/kademlia/node.cpp

```diff
@@ -668,25 +627,23 @@ void node::lookup_peers(sha1_hash const& info_hash, entry& reply
 	if (m_observer)
 		m_observer->get_peers(info_hash);
 
-	table_t::const_iterator i = m_map.lower_bound(info_hash);
-	if (i == m_map.end()) return;
-	if (i->first != info_hash) return;
+	std::string v_name;
+	std::vector<std::pair<tcp::endpoint, bool> > v_peers;
+	if (!m_storage->lookup_peers(info_hash, v_name, v_peers)) return;
```

Comment:
I was expecting this comment (and the one below).

Thinking about it, I think one solution is making the lookup_peers similar to the others. You pass the entry, and it's filled with the specified fields. I will change it and you tell me.